datahub-web-react/src/app/auth/ResetCredentials.tsx

```diff
@@ -0,0 +1,170 @@
+import { useReactiveVar } from '@apollo/client';
+import styled, { useTheme } from 'styled-components';
+import { Redirect } from 'react-router';
+import styles from './login.module.css';
+import { Message } from '../shared/Message';
+import { isLoggedInVar } from './checkAuthStatus';
+import analytics, { EventType } from '../analytics';
+import { useAppConfig } from '../useAppConfig';
+import { PageRoutes } from '../../conf/Global';
+import useGetResetTokenFromUrlParams from './useGetResetTokenFromUrlParams';
+
+type FormValues = {
+    email: string;
+    password: string;
+    confirmPassword: string;
+};
+
+const FormInput = styled(Input)`
+    &&& {
+        height: 32px;
+        font-size: 12px;
+        border: 1px solid #555555;
+        border-radius: 5px;
+        background-color: transparent;
+        color: white;
+        line-height: 1.5715;
+    }
+    > .ant-input {
+        color: white;
+        font-size: 14px;
+        background-color: transparent;
+    }
+    > .ant-input:hover {
+        color: white;
+        font-size: 14px;
+        background-color: transparent;
+    }
+`;
+
+export type ResetCredentialsProps = Record<string, never>;
+
+export const ResetCredentials: React.VFC<ResetCredentialsProps> = () => {
+    const isLoggedIn = useReactiveVar(isLoggedInVar);
+    const resetToken = useGetResetTokenFromUrlParams();
+
+    const themeConfig = useTheme();
+    const [loading, setLoading] = useState(false);
+
+    const { refreshContext } = useAppConfig();
+
+    const handleResetCredentials = useCallback(
+        (values: FormValues) => {
+            setLoading(true);
+            const requestOptions = {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify({
+                    email: values.email,
+                    password: values.password,
+                    resetToken,
+                }),
+            };
+            fetch('/resetNativeUserCredentials', requestOptions)
+                .then(async (response) => {
+                    if (!response.ok) {
+                        const data = await response.json();
+                        const error = (data && data.message) || response.status;
+                        return Promise.reject(error);
+                    }
+                    isLoggedInVar(true);
+                    refreshContext();
+                    analytics.event({ type: EventType.ResetCredentialsEvent });
+                    return Promise.resolve();
+                })
+                .catch((error) => {
+                    message.error(`Failed to log in! ${error}`);
+                })
+                .finally(() => setLoading(false));
+        },
+        [refreshContext, resetToken],
+    );
+
+    if (isLoggedIn && !loading) {
+        console.log('Is logged in');
+        return <Redirect to={`${PageRoutes.ROOT}`} />;
+    }
+
+    return (
+        <div className={styles.login_page}>
```

Comment:
For the future - we can probably reuse some of this code as it seems like much of it from the sign up page